### PR TITLE
Refactor Fridgify_Frontend

### PR DESCRIPTION
**Pull request by DennisRein**
- Remove current state
- Add repository structure
- Add communication with backend
- Add real MVC structure
- Add better logging
- Add exceptions

Usage of repositories can be found in main

Alles bis auf fridgify/lib/ ist autogenerated also nur der code da drinnen ist relevant.

### CODE REVIEW

File: fridgify/lib/data/content_repository.dart [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  ContentRepository(this.pref, this.fridge) {
+    contentApi = "${Repository.baseURL}fridge/content/${this.fridge.fridgeId}/";
+  }
+
+  @override
+  Future<int> add (Content content) async {
+    var token = getToken();
+    var body = jsonEncode({
+      "name": content.item.name, "description": content.item.description,
+      "buy_date": "2020-04-04", "expiration_date": content.expirationDate,
```
Comment (DonkeyCo) [inline fridgify/lib/data/content_repository.dart:38]: Is there a reason for the hard coded date?

File: fridgify/lib/data/content_repository.dart [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      "amount": content.amount,
+      "unit": content.unit, "store": content.item.store.name,
+    });
+    logger.i('ContentRepository => Requesting $contentApi with $body}');
```
Comment (JoschuaGoetz) [inline fridgify/lib/data/content_repository.dart:42]: I guess you missed the '}' or has that curly bracket any use ?

File: fridgify/lib/data/content_repository.dart (hunk 1 above)
Comment (DennisRein) [reply fridgify/lib/data/content_repository.dart]: Thank you pointing these things out. As seen in commit ba9f2fa, I out sourced the getToken function. I also fixed the hard coded date.

About the .gitignores. Flutter, or more so Android Studios, auto generates a .gitignore file for each sub directory to have a better overview and to be able to easier select specific directories to get ignores.

File: fridgify/lib/data/content_repository.dart (hunk 2 above)
Comment (DennisRein) [reply fridgify/lib/data/content_repository.dart]: Thanks for noticing that, I removed the forgotten '}'.

I also implemented a getHeader function as seen in 50bb59b for easier header usage/generation.

**Pull request merged**